aptos-move/move-examples/token_objects/guild/sources/guild.move

```diff
@@ -0,0 +1,309 @@
+        token: Object<MemberToken>,
+    ) acquires MemberToken {
+        let belonging_guild = borrow_global<MemberToken>(object::object_address(&token)).guild;
+        assert!(object::owner(belonging_guild) == signer::address_of(guild_master), ENOT_OWNER);
+        let member_token = move_from<MemberToken>(object::object_address(&token));
+        let MemberToken {
+            guild: _,
+            burn_ref,
+        } = member_token;
+        token::burn(burn_ref);
+    }
+
+    /// Creates the manager object.
+    fun create_manager(sender: &signer) {
+        let constructor_ref = object::create_object(signer::address_of(sender));
+        let extend_ref = object::generate_extend_ref(&constructor_ref);
+        move_to(sender, Manager { extend_ref });
+    }
+
+    /// Returns the signer of the manager object.
+    fun manager_signer(): signer acquires Manager {
+        let manager = borrow_global<Manager>(@token_objects);
+        object::generate_signer_for_extending(&manager.extend_ref)
+    }
+
+    /// Returns the address of the manager object.
+    fun manager_address(): address acquires Manager {
+        let manager = borrow_global<Manager>(@token_objects);
+        object::address_from_extend_ref(&manager.extend_ref)
+    }
+
+    /// Creates the guild collection. This function creates a collection with unlimited supply using
+    /// the module constants for description, name, and URI, defined above. The royalty configuration
+    /// is skipped in this collection for simplicity.
+    fun create_guild_collection(admin: &signer) {
+        // Constructs the strings from the bytes.
+        let description = string::utf8(GUILD_COLLECTION_DESCRIPTION);
+        let name = string::utf8(GUILD_COLLECTION_NAME);
+        let uri = string::utf8(GUILD_COLLECTION_URI);
+
+        // Creates the collection with unlimited supply and without establishing any royalty configuration.
+        collection::create_unlimited_collection(
+            admin,
+            description,
+            name,
+            option::none(),
+            uri,
+        );
+    }
+
+    /// Creates the member collection. This function creates a collection with unlimited supply using
+    /// the module constants for description, name, and URI, defined above. The royalty configuration
+    /// is skipped in this collection for simplicity.
+    fun create_member_collection(guild_token_object_signer: &signer, name: String, description: String, uri: String) {
+        // Creates the collection with unlimited supply and without establishing any royalty configuration.
+        collection::create_unlimited_collection(
+            guild_token_object_signer,
+            description,
+            name,
+            option::none(),
+            uri,
+        );
+    }
+
+    inline fun is_whitelisted(guild_master: address): bool {
+        let whitelist = &borrow_global<Config>(@token_objects).whitelist;
+        vector::contains(whitelist, &guild_master)
```

Comment:
Yes, it can. I chose `vector` for simplicity. For a large list, it's recommended to use `table`. Please see a more realistic whitelist example: https://github.com/aptos-labs/aptos-core/pull/8484/files/d48a9a73ab3f1c2154e8f4711edf507ca2d92ae4..0b54959f4393c51bc0fbe511f84def33d384a445#diff-13c7dc164d67b29d8d16d95acc90548d4850a0f2cffca51f9805e7b0f8f5a033R38